test.js

```diff
@@ -1230,3 +1231,70 @@ describe('concurrency', function() {
     expect(ParallelApi.jobs).to.equal(17);
   });
 });
+
+describe('getBabelVersion()', function() {
+  it ('returns the correct version', function() {
+    var babelCorePath = path.join(__dirname, 'node_modules', 'babel-core', 'index.js');
+    var babelCorePackageJson = path.join(path.dirname(babelCorePath), 'package.json');
+    var packageJson = fs.readFileSync(babelCorePackageJson);
+    var expectedVersion = JSON.parse(packageJson).version;
+    expect(ParallelApi.getBabelVersion()).to.equal(expectedVersion);
+  });
+});
+
+describe('workerpool', function() {
+  var parallelApiPath = require.resolve('./lib/parallel-api');
+
+  var stringToTransform = "const x = 0;";
+
+  var options = {
+    inputSourceMap: false,
+    sourceMap: false,
+    plugins: [
+      'transform-strict-mode',
+      'transform-es2015-block-scoping'
+    ]
+  };
+
+  before(function() {
+    // shut down any workerpool that is running at this point
+    var babelCoreVersion = ParallelApi.getBabelVersion();
+    var something = 'v1/broccoli-babel-transpiler/workerpool/babel-core-' + babelCoreVersion;
+    var runningPool = process[something];
```

Comment:
rather the cleaning up when the next test task starts, we should always cleanup when the current test task ends.
This ensures our tests leave themselves in a pristine state.

So specifically, the `terminate` and `delete` should be in `after` or `afterEach`

Reply:
moved the workerpool cleanup to after() for the tests that use it